WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogViewModel.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+import Combine
+import UIKit
+
+struct ApplicationLogLine: Identifiable {
+    let id = UUID()
+
+    let date: Date?
+    let text: String
+
+    private let dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "YYYY-MM-dd HH:mm:ss:SSS"
+        return dateFormatter
+    }()
+
+    private let datePrefixPattern = try! NSRegularExpression(pattern: "^(\\d{4}/\\d{2}/\\d{2} \\d{2}:\\d{2}:\\d{2}\\:\\d{3})  (.*)")
```

Comment:
⚠️  What is this pattern mattiching? 
Also, could we get rid of the `try!` 😅

Reply:
Added some annotation in a724eda62.

`try!` and force unwrapping always makes me uncomfortable, but I think it's sometimes OK. As long as unit tests are passing, this is very unlikely to fail, and if it does, a crash will alert us about it quickly. I see this type of thing comparable to the force-unwrapping we do in our image extensions in `UIImage+Woo`.

The alternative I see used in other places in the app would be to ignore the error, maybe log a message, and act as if the date was invalid instead of the regular expression. This might not crash, but I imagine it could be months before we realized something was off.